.travis/check_cubemx_ci.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+#
+# This script is used to check for corrupt ioc files in CI
+#
+
+# The directory this script is in
+CURR_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+echo "=========================================================="
+echo " THIS SHOULD ONLY BE RUN IN CI, PLEASE DO NOT USE LOCALLY"
+echo "=========================================================="
+
+cd src
+bazel run --cpu=stm32h7 //firmware_new/tools:cubemx_regen firmware_new/boards/frankie_v1
+if [[ "$?" != 0 ]]; then
+    echo "There was a problem running the formatting script, stopping now."
+    exit 1
+fi
+
+cd ..
+# Fix formatting
+./formatting_scripts/fix_formatting.sh
```

Comment:
you want to run `git diff` after this command to see if the code was generated correctly

we care if there was a diff after `fix_formatting.sh`, the output of `fix_formatting.sh` itself we don't really care about

Reply:
git diff exits with 0 if there is no diff, so you can keep the same check after